avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGenerator.java

```diff
@@ -633,11 +635,20 @@ private void processUnion(JVar unionSchemaVar, final String name, final Schema u
 
         if (null == readerOptionSchema) {
           // This is the same exception that vanilla Avro would throw in this circumstance
-          thenBlock._throw(JExpr._new(codeModel.ref(AvroTypeException.class)).arg(
-              JExpr.lit("Found " + optionSchema + ", expecting " + (readerSchemaNotAUnion ? unionReaderSchema.toString()
-                      : unionReaderSchema.getTypes().toString()))));
+          String fullExceptionString = "Found " + optionSchema + ", expecting " + (readerSchemaNotAUnion ? unionReaderSchema.toString()
+              : unionReaderSchema.getTypes().toString());
+
+          if (fullExceptionString.length() <= MAX_LENGTH_OF_STRING_LITERAL) {
```

Comment:
Can we fold this condition check into the newly created function: `composeLargeStringLiteral`? and rename it to be `composeStringLiteral` since it can handle both small and large strings.

Reply:
Done